ui/js/graphs.js

```diff
@@ -599,9 +617,15 @@ perf.controller('GraphsCtrl', [
 
           // we only store the signature + project name in the url, we need to
           // fetch everything else from the server
-          var partialSeriesList = $stateParams.series.map(function(encodedSeries) {
-            return JSON.parse(decodeURIComponent(encodedSeries));
-          });
+          
+          var partialSeriesString = decodeURIComponent($stateParams.series.pop());
+          partialSeriesString = partialSeriesString.replace(/[\[\]"]/g, ''); 
+          var partialSeriesArray = partialSeriesString.split(",");
+          var partialSeriesList = Object.create(null)
+          partialSeriesList["project"] = partialSeriesArray[0]
+          partialSeriesList["signature"] = partialSeriesArray[1]
+          partialSeriesList["visible"] = (partialSeriesArray[2]) ? partialSeriesArray[2] : true 
+          partialSeriesList = $.makeArray(partialSeriesList)
```

Comment:
I'm not sure if I understand this 100%, but I don't think this will work if there is more than one graph specified in the UI?